repose-aggregator/commons/commons-utilities/src/main/scala/org/openrepose/commons/utils/servlet/http/HttpServletRequestWrapper.scala

```diff
@@ -145,4 +144,120 @@ class HttpServletRequestWrapper(originalRequest: HttpServletRequest, inputStream
   def getSplittableHeaderScala(headerName: String): List[String] = getHeadersScala(headerName).foldLeft(List.empty[String])((list, s) => list ++ s.split(","))
 
   override def getSplittableHeaders(headerName: String): util.List[String] = getSplittableHeaderScala(headerName).asJava
+
+  /**
+    * @return a [[StringBuffer]] containing the reconstructed URL for this request (note that
+    *         mutation of this [[StringBuffer]] will have no effect on the request URL)
+    */
+  override def getRequestURL: StringBuffer = {
+    val url: StringBuffer = new StringBuffer(getScheme).append("://").append(getServerName)
+
+    if (getServerPort > 0 && ((HTTP.equalsIgnoreCase(getScheme) && getServerPort != 80) || (HTTPS.equalsIgnoreCase(getScheme) && getServerPort != 443))) {
+      url.append(':').append(getServerPort)
+    }
+
+    if (Option(getRequestURI).exists(_.nonEmpty)) {
+      url.append(getRequestURI)
+    }
+
+    url
+  }
+
+  override def getRequestURI: String = requestUri
+
+  def setRequestURI(uri: String): Unit = {
+    if (Option(uri).isEmpty) throw new IllegalArgumentException("null is not a legal argument to setRequestURI")
+
+    requestUri = uri
+  }
+
+  /**
+    * @return a string representation of the query parameters for this request
+    */
+  override def getQueryString: String = queryString
+
+  /**
+    * @param newQueryString the desired query string for this request
+    */
+  def setQueryString(newQueryString: String): Unit = {
+    def parseQueryString(s: String): Map[String, Array[String]] = {
+      val parameterMap = mutable.Map.empty[String, Array[String]]
+
+      s.split(QueryPairDelimiter) foreach { queryPair =>
+        val keyValuePair = queryPair.split(QueryKeyValueDelimiter, 2)
+
+        if (keyValuePair.length == 2) {
+          val key = keyValuePair(0)
+          val value = keyValuePair(1)
+          parameterMap += (key -> parameterMap.getOrElse(key, Array.empty[String]).:+(value))
+        } else {
+          val key = keyValuePair(0)
+          parameterMap += (key -> parameterMap.getOrElse(key, Array.empty[String]).:+(""))
+        }
+      }
+
+      parameterMap.toMap
+    }
+
+    val updatedParameterMap = mutable.Map.empty[String, Array[String]]
+    val curQueryMap = Option(getQueryString).map(parseQueryString).getOrElse(Map.empty[String, Array[String]])
+    val newQueryMap = Option(newQueryString).map(parseQueryString).getOrElse(Map.empty[String, Array[String]])
+
+    // Remove all current query parameters from the parameter map
+    getParameterMap.asScala foreach { case (key, values) =>
+      val formValues = values.filterNot(value => curQueryMap.get(key).exists(_.contains(value)))
```

Comment:
this is overly aggressive. it'll remove the form value as well as the query value if they have the same value.